spec/operators/publishReplay-spec.ts

```diff
@@ -409,4 +409,75 @@ describe('Observable.prototype.publishReplay', () => {
 
     published.connect();
   });
+
+  it('should mirror a simple source Observable with selector', () => {
+    const values = {a: 2, b: 4, c: 6, d: 8};
+    const selector = observable => observable.map(v => 2 * v);
+    const source = cold('--1-2---3-4---|');
+    const sourceSubs =  '^             !';
+    const published = source.publishReplay(1, Number.POSITIVE_INFINITY, selector);
+    const expected =    '--a-b---c-d---|';
+
+    expectObservable(published).toBe(expected, values);
+    expectSubscriptions(source.subscriptions).toBe(sourceSubs);
+  });
+
+  it('should emit an error when the selector throws an exception', () => {
+    const error = "It's broken";
+    const selector = () => {
+      throw error;
+    };
+    const source = cold('--1-2---3-4---|');
+    const published = source.publishReplay(1, Number.POSITIVE_INFINITY, selector);
+
+    // The exception is thrown outside Rx chain (not as an error notification).
+    expect(() => published.subscribe()).to.throw(error);
```

Comment:
This exception is thrown outside the chain so I need to check it like this. This situations isn't tested in neither `publish` nor `multicast` (if I didn't miss anything) so I'm not sure I'm doing it right.

Reply:
OH GEEZ! I missed this... So what *should* happen is you should catch the error and send it down the `error` path with `observer.error(err)`. There are all sorts of examples of how to handle this around teh library. 

Basically any user-supplied function needs to have it's errors caught and sent down the error channel of observation.

Reply:
But I would expect `multicast` to catch this error for me because that's where the selector is called: https://github.com/ReactiveX/rxjs/blob/master/src/operators/multicast.ts#L64